src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
@kapilash I'm having trouble following this conversation in GitHub.  I don't see 155-158.  Can you please paste in the revised proposed code to show how you are implementing the 'getter' and 'setter' for `UseProxy`.